core/commands/dns.go

```diff
@@ -63,7 +63,7 @@ The resolver will give:
 			depth = namesys.DefaultDepthLimit
 		}
 		output, err := resolver.ResolveN(req.Context().Context, name, depth)
-		if err != nil {
+		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
```

Comment:
From an IRC discussion with @jbenet on 2015-06-18:

16:14 <jbenet> i think the dns --recursive can just: resolve domains until it can't any more.
16:14 <jbenet> if it ends in /ipns/... cool. if it ends in /ipfs/... even cooler.
16:15 <wking> and if it ends in /tiddlywinks?
16:15 <jbenet> that's cool too.
16:16 <jbenet> the "dns resolver" only promised to resolve dns
16:16 <wking> how about for other resolvers?
16:16 <wking> E.g. if /ipns/QmSomeHash resolves to /tiddlywinks?
16:16 <jbenet> each should resolve only what they can
16:17 <jbenet> thats what the top level multi-protocol resolver is there to do
16:17 <wking> ok, so we just pull out the /ipfs/ special-case, and have callers handle that (if they expect an /ipfs/ entry)
16:17 <wking> that sounds good to me
16:17 <jbenet> yep